akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -134,6 +134,7 @@ private[http] object HttpServerBluePrint {
 
           val entity = createEntity(entityCreator) withSizeLimit settings.parserSettings.maxContentLength
           val httpRequest = HttpRequest(effectiveMethod, uri, effectiveHeaders, entity, protocol)
+            .withAttributes(attrs)
```

Comment:
I guess this might have a small performance impact. In that particular case, we can probably optimize `withAttributes` to be a no-op for `Map.empty` but we should also monitor if we should introduce a private constructor that includes attributes as well to avoid extra copying in cases like this.